apps/gisday-angular/src/environments/environment.dev.ts

```diff
@@ -0,0 +1,20 @@
+// This file can be replaced during build by using the `fileReplacements` array.
+// `ng build --prod` replaces `environment.ts` with `environment.prod.ts`.
+// The list of file replacements can be found in `angular.json`.
+
+export const environment = {
+  production: false
+};
+
+/*
+ * For easier debugging in development mode, you can import the following file
+ * to ignore zone related error stack frames such as `zone.run`, `zoneDelegate.invokeTask`.
+ *
+ * This import should be commented out in production mode because it will have a negative impact
+ * on performance if an error is thrown.
+ */
+// import 'zone.js/plugins/zone-error';  // Included with Angular CLI.
+
+export const api_url = 'http://localhost:3333/api';
+export const idp_url = 'http://localhost:4001/oidc';
```

Comment:
What's wrong with having the option, especially inside an environment file, to use either a local IdP or a hosted one?

Reply:
Having vs not having an option is not the problem. I just didn't see any use of `idp_url` anywhere, so i can't tell where in the code it's meant to use a local vs hosted one. I did a quick search and found nothing so I *figured* since it's not called anywhere, it's not actually being used.

Reply:
It *can* be used inside of app.module.ts as the authority (IdP URL) value. Sometimes I wanna use a my local IdP.